vault/diagnose/tls_test.go

```diff
@@ -132,15 +87,183 @@ func TestTLSTrailingData(t *testing.T) {
 	}
 }
 
-func TestTLSExpiredCert(t *testing.T) {
+func TestTLSExpiredCert(t *testing.T) {}
+
+// TestTLSMismatchedCryptographicInfo verifies that a cert and key of differing cryptographic
+// types, when specified together, is met with a unique error message.
+func TestTLSMismatchedCryptographicInfo(t *testing.T) {
+	listeners := []listenerutil.Listener{
+		{
+			Config: &configutil.Listener{
+				Type:                          "tcp",
+				Address:                       "127.0.0.1:443",
+				ClusterAddress:                "127.0.0.1:8201",
+				TLSCertFile:                   "./../../api/test-fixtures/keys/cert.pem",
+				TLSKeyFile:                    "./test-fixtures/ecdsa.key",
+				TLSClientCAFile:               "./../../api/test-fixtures/root/rootcacert.pem",
+				TLSMinVersion:                 "tls10",
+				TLSRequireAndVerifyClientCert: true,
+				TLSDisableClientCerts:         false,
+			},
+		},
+	}
+	err := ListenerChecks(listeners)
+	if err == nil {
+		t.Errorf("TLS Config check on fake certificate should fail")
```

Comment:
nit: this could be `t.Error()`

Reply:
Will do (for all of these errors that don't have formatting).